test_runner/regress/test_broken_timeline.py

```diff
@@ -100,22 +101,3 @@ def test_create_multiple_timelines_parallel(neon_simple_env: NeonEnv):
         ]
         for future in futures:
             future.result()
-
-
-def test_fix_broken_timelines_on_startup(neon_simple_env: NeonEnv):
-    env = neon_simple_env
-
-    tenant_id, _ = env.neon_cli.create_tenant()
-
-    # Introduce failpoint when creating a new timeline
-    env.pageserver.safe_psql("failpoints before-checkpoint-new-timeline=return")
-    with pytest.raises(Exception, match="before-checkpoint-new-timeline"):
-        _ = env.neon_cli.create_timeline("test_fix_broken_timelines", tenant_id)
-
-    # Restart the page server
-    env.neon_cli.pageserver_stop(immediate=True)
-    env.neon_cli.pageserver_start()
-
-    # Check that tenant with "broken" timeline is not loaded.
-    with pytest.raises(Exception, match=f"Failed to get repo for tenant {tenant_id}"):
-        env.neon_cli.list_timelines(tenant_id)
```

Comment:
What happens now in this scenario?

Reply:
I think you might be commenting some old change, now this test is called `test_fix_broken_timelines_on_startup` and does more or less the same.

By test's scenario, we've spoiled the timeline (failpoint caused it to crash after only tmp directory is created). 

Before, we would have failed the entire tenant due to one of its timeline failing to load. This causes HTTP queries to fail, since failed tenants in the old code are not loaded into the memory.

Now, we load the tenant successfully, since all its files are intact. 
Then, we try to load the timeline and remove all of its tmp files, failing to load the timeline and not attaching it to the tenant, marking it "broken".
This way, the tenant is still in the memory and does not cause odd errors when it's attempted to get re-initialized, but does not show any "errored" timelines in it, and this is also supposed to change later, when I apply similar logic to the timeline loading: timelines, that fail to load from disk during init, should still be present in memory with some different status.